azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java

```diff
@@ -379,6 +380,10 @@ protected IFunctionAppDeploymentSlot updateDeploymentSlot(final IFunctionAppDepl
     }
 
     private void deployArtifact(IFunctionAppBase target) throws AzureExecutionException {
+        if (target.getRuntime().getOperatingSystem() == OperatingSystem.DOCKER) {
```

Comment:
shall we check target.getRuntime()  == null?

Reply:
target.getRuntime() should not be null, I may add nonnull annotation later